src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -1204,16 +1156,18 @@ CreateForeignConstraintsToReferenceTable(List *logicalRepTargetList)
 		 */
 		foreach_ptr(shardInterval, target->newShards)
 		{
-			List *commandList = GetForeignConstraintCommandsToReferenceTable(
+			List *commandList = CopyShardForeignConstraintCommandList(
 				shardInterval);
+			commandList = list_concat(
+				list_make1("SET LOCAL citus.skip_constraint_validation TO ON;"),
+				commandList);
 
-			char *command = NULL;
+			SendCommandListToWorkerOutsideTransaction(
```

Comment:
I've noticed that the constant opening and closing of connections, especially in splits, has started weighing in on how long we block writes.

Reply:
Yeah, I wasn't super happy about it either. In a follow up PR I'll make sure we can reuse the existing connection.